pkg/plugins/policies/meshaccesslog/plugin/v1alpha1/plugin.go

```diff
@@ -24,7 +31,78 @@ func (p plugin) MatchedPolicies(dataplane *core_mesh.DataplaneResource, resource
 	return matchers.MatchedPolicies(api.MeshAccessLogType, dataplane, resources)
 }
 
+func (p plugin) applyToOutbounds(rs *core_xds.ResourceSet, proxy *core_xds.Proxy) error {
+	logPolicies, ok := proxy.Policies.Dynamic[api.MeshAccessLogType]
+	if !ok {
+		return nil
+	}
+
+	sourceService := proxy.Dataplane.Spec.GetIdentifyingService()
+
+	outboundListeners := map[mesh_proto.OutboundInterface]*envoy_listener.Listener{}
+	for _, res := range rs.Resources(envoy_resource.ListenerType) {
+		if res.Origin != generator.OriginOutbound {
+			continue
+		}
+		listener := res.Resource.(*envoy_listener.Listener)
+
+		address := listener.GetAddress().GetSocketAddress()
+
+		outboundListeners[mesh_proto.OutboundInterface{
+			DataplaneIP:   address.GetAddress(),
+			DataplanePort: address.GetPortValue(),
+		}] = listener
```

Comment:
Don't get tricked by this it contains extra fields. @lobkovilya introduced another type which has just IP and Port and makes this less error prone

Reply:
Yeah, but I called it [InboundListener](https://github.com/kumahq/kuma/blob/master/pkg/core/xds/matched_policies.go#L38), @michaelbeaumont feel free to rename it to `Listener`